extensions/authentication-passport/src/strategy-adapter.ts

```diff
@@ -44,8 +50,16 @@ export class StrategyAdapter implements AuthenticationStrategy {
       const strategy = Object.create(this.strategy);
 
       // add success state handler to strategy instance
-      strategy.success = function(user: UserProfile) {
-        resolve(user);
+      // as a generic adapter, it is agnostic of the type of
+      // the custom user, so loose the type restriction here
+      // to be `any`
+      strategy.success = function(user: any) {
+        if (self.userConverter) {
```

Comment:
Let's supply a default converter function in the constructor, so that we don't have to check if `self.userConverter` is set.

```ts
// In the constructor
constructor(
    private readonly strategy: Strategy, 
    readonly name: string,
    private userConverter: convertUserToUserProfileFn<U> = u => u
  ) {}

// usage
const userProfile = self.userConverter(user);
resolve(userProfile);
```